### PR TITLE
Add possibility to set filename.

### PR DESCRIPTION
**Pull request by Primajin**
Refers to #137 

@garris If you want the commits squashed I can do that beforehand, but it would also be possible to just use the github squash Button:

Image of Squash and Merge

https://help.github.com/articles/about-pull-request-merge-squashing/

Which one would you prefer?

### CODE REVIEW

**Comment by garris** [comment]
I forgot about that squish feature -- thanks!  Will try to run this today.

**Comment by Primajin** [comment]
🎉  Great!

File: README.md [hunk 1]
```diff
@@ -439,6 +439,27 @@ Please note: these file paths are relative to your `./node_modules/backstopjs/`
   }
 ```
 
+### Setting custom reference file names (version 1.3.0+)
```
Comment (Pofigizm) [inline README.md]: Don't forget change to correct version here.

**Comment by Primajin** [comment]
Thanks @pofigizm for the reminder....actually I lost  a bit track of the version counting...maybe @garris can help out here and say when its time to go for major, minor or patch.

😉

**Comment by garris** [comment]
@Primajin thanks so much for the feature work!  I had just one comment which I left on the PR.   please just address that and this feature will be merged for the next release.  

FYI: there is another feature #249 (selector expansion) which would effect behavior of the filenaming (downstream of your naming template).   Please feel free to make any comments, suggestions etc as this would dynamically add selector names for selectors with multiple instances.

Cheers.

**Comment by Primajin** [comment]
Hey @garris , great! Just one thing

> I had just one comment which I left on the PR. please just address that [...]

Sorry I don't get what you mean. Where is this comment? Can you point me there. Is it in #137 or in here? Is it a line comment or a PR note? I will work on that asap and will also have a look at #249 !

Best
Jannis / @Primajin

File: capture/genBitmaps.js [hunk 2]
```diff
@@ -12,12 +11,25 @@ if (!config.paths) {
   config.paths = {};
 }
 
+String.prototype.hashCode = function () {
```
Comment (garris) [inline capture/genBitmaps.js]: What is the reason for adding this to the string prototype as opposed to just a local method?

**Comment by garris** [comment]
@Primajin sorry, the comment is here ^^^

File: capture/genBitmaps.js (hunk 2 above)
Comment (Primajin) [inline capture/genBitmaps.js]: Well it was "lented" from here: http://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/ ;)

**Comment by garris** [comment]
Just an update.  I am going to merge this into the version_2_0_RC branch.  Version 2 is a huge refactor (completely removed gulp) and this feature is required for that one.   I'll ping everyone when it's in to give people a chance to test.   Thank you very much for this feature work  -- I appreciate your persistence.   Cheers.

**Comment by onigoetz** [comment]
By the way, we already changed the default file name in this new branch : https://github.com/onigoetz/BackstopJS/blob/1f8d7a6c95264dbf5f95fbf5d37c6bacf5405413/capture/genBitmaps.js#L173

We use the label of a scenario to generate the name and not the scenario's index.

**Comment by matthew-dean** [comment]
How do you actually change the reference file name using this feature? o_O I can't find any existing documentation to do so.

**Comment by garris** [comment]
It is quite easy — however — not recommended since backstop requires enough unique information in the filename to manage everything.    That said, if you’re careful you can probably get it to work just fine.

Any case, check the readme here... https://github.com/garris/BackstopJS/blob/master/README.md#if-you-just-upgraded-to-2x-or-3x

**Comment by matthew-dean** [comment]
Yep, I figured it out but forgot to find this thread to update you, sorry about that.

**Pull request merged**